clusterloader2/api/types.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"time"
+)
+
+// Config is a structure that represents configuration
+// for a single test scenario.
+type Config struct {
+	// Namespaces is a number of automanged namespaces.
+	Namespaces int32 `json: namespaces`
+	// Steps is a sequence of test steps.
+	// Steps will be executed one at a time, in order specified by the position in array.
+	Steps []Step `json: steps`
+	// TuningSets is a collection of tuning sets that can be used by steps.
+	TuningSets []TuningSet `json: tuningSets`
+}
+
+// Step is a structure that represents encapsulation
+// of some actions. This actions could be object declarations
+// or measurement usages.
+// Exactly one structure (Phases or Measurments) should be non-empty.
+type Step struct {
+	// Phases is a collection of declarative definitions of objects.
+	// Phases will be executed in parallel.
+	Phases []Phase `json: phases`
+	// Measurements is a collection of measurement calls.
+	Measurements []Measurement `json: measurements`
+}
+
+// Phase is a structure that declaratively defines state of objects.
+// In a given namespace range (or cluster scope if no range is spcecified)
+// it definies the number and the configuration of managed objects.
+type Phase struct {
+	// NamespaceRange defines the set of namespaces in which objects
+	// should be reconciled.
+	// If null, objects are assumed to be cluster scoped.
+	NamespaceRange *NamespaceRange `json: namespaceRange`
+	// ReplicasPerNamespace is a number of instances of a given object
+	// to exist in each of referenced namespaces.
+	ReplicasPerNamespace int32 `json: replicasPerNamespace`
+	// TuningSet is the name of TuningSet to be used.
+	TuningSet string `json: tuningSet`
+	// Objects is a set of objects that should be reconciled.
+	Objects []Object `json: objects`
+}
+
+// Object is a structure that defines the object managed be the tests.
+type Object struct {
+	// ObjectType is a type definition for a given object.
```

Comment:
ObjectType defines a type of a given object.

Reply:
done